web/annotation_layer_builder.js

```diff
@@ -44,88 +40,73 @@ var AnnotationLayerBuilder = (function AnnotationLayerBuilderClosure() {
+        renderInteractiveForms: this.renderInteractiveForms,
+        linkService: this.linkService,
+        downloadManager: this.downloadManager,
       };
 
-      this.pdfPage.getAnnotations(parameters).then(function (annotations) {
-        viewport = viewport.clone({ dontFlip: true });
-        parameters = {
-          viewport: viewport,
-          div: self.div,
-          annotations: annotations,
-          page: self.pdfPage,
-          renderInteractiveForms: self.renderInteractiveForms,
-          linkService: self.linkService,
-          downloadManager: self.downloadManager,
-        };
-
-        if (self.div) {
-          // If an annotationLayer already exists, refresh its children's
-          // transformation matrices.
-          AnnotationLayer.update(parameters);
-        } else {
-          // Create an annotation layer div and render the annotations
-          // if there is at least one annotation.
-          if (annotations.length === 0) {
-            return;
-          }
+      if (this.div) {
+        // If an annotationLayer already exists, refresh its children's
+        // transformation matrices.
+        AnnotationLayer.update(parameters);
+      } else {
+        // Create an annotation layer div and render the annotations
+        // if there is at least one annotation.
+        if (annotations.length === 0) {
+          return;
+        }
 
-          self.div = document.createElement('div');
-          self.div.className = 'annotationLayer';
-          self.pageDiv.appendChild(self.div);
-          parameters.div = self.div;
+        this.div = document.createElement('div');
+        this.div.className = 'annotationLayer';
+        this.pageDiv.appendChild(this.div);
+        parameters.div = this.div;
 
-          AnnotationLayer.render(parameters);
-          if (typeof mozL10n !== 'undefined') {
-            mozL10n.translate(self.div);
-          }
+        AnnotationLayer.render(parameters);
+        if (typeof mozL10n !== 'undefined') {
+          mozL10n.translate(this.div);
         }
-      });
-    },
-
-    hide: function AnnotationLayerBuilder_hide() {
-      if (!this.div) {
-        return;
       }
-      this.div.setAttribute('hidden', 'true');
-    }
-  };
+    });
+  }
 
-  return AnnotationLayerBuilder;
-})();
+  hide() {
+    if (!this.div) {
+      return;
+    }
+    this.div.setAttribute('hidden', 'true');
+  }
+}
 
 /**
- * @constructor
  * @implements IPDFAnnotationLayerFactory
  */
-function DefaultAnnotationLayerFactory() {}
-DefaultAnnotationLayerFactory.prototype = {
+class DefaultAnnotationLayerFactory {
   /**
    * @param {HTMLDivElement} pageDiv
    * @param {PDFPage} pdfPage
    * @param {boolean} renderInteractiveForms
    * @returns {AnnotationLayerBuilder}
    */
-  createAnnotationLayerBuilder: function (pageDiv, pdfPage,
-                                          renderInteractiveForms) {
+  createAnnotationLayerBuilder(pageDiv, pdfPage, renderInteractiveForms) {
```

Comment:
Done in the new commit. I have updated the interface as well. With the new ES6 syntax, ESLint is able to detect that the interface is not used anywhere (expected, because it's an interface after all), so we have to disable linting there. Otherwise it's a trivial change.